CMakeLists.txt

```diff
@@ -135,9 +135,30 @@ ADD_CUSTOM_TARGET(setup_tests
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
   COMMENT "Enabling all tests ...")
 
+SET(ASPECT_TEST_GENERATOR "Unix Makefiles" CACHE STRING "Generator to use for the test cmake project. Using ninja instead of make is not recommended.")
+
 IF(EXISTS ${CMAKE_SOURCE_DIR}/tests/CMakeLists.txt)
-  ENABLE_TESTING()
-  ADD_SUBDIRECTORY(tests)
+  FILE(WRITE ${CMAKE_BINARY_DIR}/CTestTestfile.cmake "SUBDIRS(tests)")
+
+  FILE(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/tests)
+
+  SET(aspect_binary ${CMAKE_BINARY_DIR}/aspect)
+
+  MESSAGE(STATUS "Setting up test project, see setup_tests.log for details.")
+  EXECUTE_PROCESS(
+    COMMAND ${CMAKE_COMMAND} -G ${ASPECT_TEST_GENERATOR}
```

Comment:
in your example, `ASPECT_TEST_GENERATOR` is a string with spaces. What happens when it is passed like this? Will the command line take this apart into separate strings?